### PR TITLE
refactor(toml): Rename 'resolved' to 'normalized'

### PR DESCRIPTION
**Pull request by epage**
In a discussion on an issue, it became confusing to talk about "resolved" manifests and dependency resolution,
so I'm switching manifests to use the other term I considered, "normalized".

### CODE REVIEW

**Comment by weihanglo** [comment]
> In a discussion on an issue

Which discussion?

(It might be me but I forgot…)

**Comment by epage** [comment]
I don't remember.

**Comment by weihanglo** [comment]
https://github.com/rust-lang/cargo/pull/13589#discussion_r1526632165

Might not be this but it's the closest one I could find.

File: src/cargo/util/toml/mod.rs [hunk 1]
```diff
@@ -314,52 +314,52 @@ fn resolve_toml(
...
         let package_name = &original_package.name;
 
-        let resolved_package =
+        let normalized_package =
             resolve_package_toml(original_package, features, package_root, &inherit)?;
-        let edition = resolved_package
-            .resolved_edition()
+        let edition = normalized_package
+            .normalized_edition()
             .expect("previously resolved")
```
Comment (weihanglo) [inline src/cargo/util/toml/mod.rs:321]:
```suggestion
            .expect("previously normalized")
```

File: src/cargo/util/toml/mod.rs [hunk 2]
```diff
@@ -240,28 +240,28 @@ fn to_workspace_config(
...
-        &resolved_toml.metadata,
+        &normalized_toml.metadata,
     );
     ws_root_config
 }
 
-/// See [`Manifest::resolved_toml`] for more details
+/// See [`Manifest::normalized_toml`] for more details
 #[tracing::instrument(skip_all)]
 fn resolve_toml(
```
Comment (weihanglo) [inline src/cargo/util/toml/mod.rs:266]: Should we update function names?

File: src/cargo/util/toml/mod.rs [hunk 3]
```diff
@@ -1770,15 +1776,15 @@ struct ManifestContext<'a, 'b> {
...
     kind: Option<DepKind>,
 ) -> CargoResult<()> {
-    let Some(dependencies) = resolved_deps else {
+    let Some(dependencies) = normalized_deps else {
         return Ok(());
     };
 
     for (n, v) in dependencies.iter() {
-        let resolved = v.resolved().expect("previously resolved");
+        let resolved = v.normalized().expect("previously resolved");
```
Comment (weihanglo) [inline src/cargo/util/toml/mod.rs:1787]:
```suggestion
        let normalized = v.normalized().expect("previously normalized");
```

File: src/cargo/util/toml/targets.rs [hunk 4]
```diff
@@ -176,12 +176,12 @@ pub fn resolve_lib(
...
-    resolved_lib: Option<&TomlLibTarget>,
+    normalized_lib: Option<&TomlLibTarget>,
     package_root: &Path,
     edition: Edition,
     warnings: &mut Vec<String>,
 ) -> CargoResult<Option<Target>> {
-    let Some(lib) = resolved_lib else {
+    let Some(lib) = normalized_lib else {
         return Ok(None);
     };
```
Comment (weihanglo) [inline src/cargo/util/toml/targets.rs:187]: Some other locations need update are out of the diff:

* `fn is_resolved`
* `expect("previously resovled")` and `expect("should be resolved")`.

File: src/cargo/core/manifest.rs [hunk 5]
```diff
@@ -483,9 +483,9 @@ impl Manifest {
     pub fn contents(&self) -> &str {
         self.contents.as_str()
     }
-    /// See [`Manifest::resolved_toml`] for what "resolved" means
+    /// See [`Manifest::normalized_toml`] for what "resolved" means
     pub fn to_resolved_contents(&self) -> CargoResult<String> {
```
Comment (weihanglo) [inline src/cargo/core/manifest.rs:487]:
```suggestion
    /// See [`Manifest::normalized_toml`] for what "normalized" means
    pub fn to_normalized_contents(&self) -> CargoResult<String> {
```

**Comment by epage** [comment]
> Might not be this but it's the closest one I could find.

It was more recent than that, like in the last month or so.  Maybe in regards to implicit features as we were talking about resolving features and resolving dependencies.

**Comment by epage** [comment]
Wow, my regexes did not catch as much as I thought

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks for the rename!

@bors r+

**Pull request merged**